### ISSUE TITLE
LaTeX support?

### ISSUE THREAD

**Issue by Tapwatero** [mdx-editor/editor#59]
Hello, I am wondering if MDX supports LaTeX markdown for mathematics - if so I would appreciate documentation or a link to such documentation as I could not find it - if not the implemenation would not go unused.

**Comment by petyosi** [comment]
You're not wrong, there's no such support at the moment. This has been requested before - I wonder how would you like to see the editing experience? To clarify what I mean, rendering the resulting latex is pretty much straightforward, but how would you like to get the editing going - like some form of a source code?

**Comment by Tapwatero** [comment]
> 

I think latex needs to be writable not as a toolbar button, so I could write the maths and when im off the line or something along those lines - it would be rendered. Let me know if this is not clear enough, I don't know how to explain it

**Comment by hahhen** [comment]
Hey, if you're still searching for it, you should try using MathLive inside the editor, I just did that. Make a component with a radix dialog, a button with the same classes as the toolbar button with an svg icon inside it and insert a <math-field> inside it. After you're done, just get the value of the math-field and insertMarkdown it into the editor.
Check their documentation on their website, it's a pretty good LaTeX editor

**Comment by tumidi** [comment]
> you should try using MathLive inside the editor, I just did that.

MathLive was the first thing coming to my mind, too. Do you happen to have an example of integrating MathLive with MDXEditor?

**Comment by hahhen** [comment]
> > you should try using MathLive inside the editor, I just did that.
> 
> MathLive was the first thing coming to my mind, too. Do you happen to have an example of integrating MathLive with MDXEditor?

Actually, yes! You can check my github, it's in
 https://github.com/hahhen/forum_teste/tree/main/src/components/editor

Actually, it's not a proper MDXEditor plugin, since it doesn't implement the plugin functionalities, but it works well enough

**Comment by mohad12211** [comment]
does anyone have an example on how to do this? the link posted above seems to be deleted

**Comment by hahhen** [comment]
https://github.com/hahhen/cpsforum/blob/master/src%2Fcomponents%2Feditor%2Finsertmathlive.jsx

**Comment by hahhen** [comment]
Refer to the mathlive documentation: renderMathInElement renders based on the element id that is sent as the function's parameter. My editor is wrapped by a div of which id is "editor". If anything else bugs, you can check my editor and sort it out, it's in the same folder as the insertMathLive

**Comment by 0x0f0f0f** [comment]
I would implement such a plugin. 
Looking at https://playground.lexical.dev/ there is an equation plugin using KaTeX which seems to work just alright.

We then need to add markdown shortcuts. It would be nice to have single wrapping `$` for inline maths, and double for full-line equations

**Comment by 0x0f0f0f** [comment]
Here's the rendering component used in Lexical playground https://github.com/facebook/lexical/blob/main/packages/lexical-playground/src/ui/KatexRenderer.tsx

**Comment by ta-sakin** [comment]
> I would implement such a plugin. Looking at https://playground.lexical.dev/ there is an equation plugin using KaTeX which seems to work just alright.
> 
> We then need to add markdown shortcuts. It would be nice to have single wrapping `$` for inline maths, and double for full-line equations

It would be useful to have a plugin with katex. Please update here if you implement one. @0x0f0f0f
